x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java

```diff
@@ -483,33 +496,77 @@ X509ExtendedTrustManager wrapWithDiagnostics(X509ExtendedTrustManager trustManag
                     .limit(2) // we only need to distinguishing between 0/1/many
                     .map(Entry::getKey)
                     .collect(Collectors.toUnmodifiableList());
+                final String name;
                 switch (names.size()) {
                     case 0:
-                        return "(unknown)";
+                        name = "(unknown)";
+                        break;
                     case 1:
-                        return names.get(0);
+                        name = names.get(0);
+                        break;
                     default:
-                        return "(shared)";
+                        name = "(shared)";
+                        break;
                 }
+                return name + " (with trust configuration: " + configuration.getTrustConfig() + ")";
```

Comment:
This is new functionality.
I realised we can supplement the name of the context with a description of the trust config.
 
The benefit of that is that when we have a `(shared)` context, we don't need to guess which shared context it might be, we can see the exact configuration.